### PR TITLE
Ignore the type in SGEN if it contains any property that only have private setter

### LINKED ISSUE #19723: Fail to generate the code if the type of the class contains any property that only have private setter.
During code generation, if any of the class contains the property that only have private setter, it will throw exception. We need ignore this type but not make the code generation fail. This also repro on desktop.

### PR DESCRIPTION
**Pull request by huanwu**
Fix #19723

@shmao @zhenlan @mconnew

### CODE REVIEW

**Comment by huanwu** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by huanwu** [comment]
@dotnet-bot test NETFX x86 Release Build

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 1]
```diff
@@ -1165,4 +1165,24 @@ public override int GetHashCode()
         return this.DoubleProp.GetHashCode() ^ this.FloatProp.GetHashCode() ^
             this.DoubleField.GetHashCode() ^ this.SingleField.GetHashCode();
     }
+}
+
+public class TypeWithoutPublicSetter
+{
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: Do we need to move the test case as well?

Comment (huanwu) [reply src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: The existing test contains many invalid type like invalidnestedtype, which will cause stackoverflow exception.

**Pull request merged**